terraform/aws/modules/elastiCache/memecached/README.md

```diff
@@ -0,0 +1,50 @@
+# ElastiCache Memcached Module
+This module will creates a fully functional Memcached cluster via Terraform. As a part of the module it will generate the following:
+
+- A Memcached cluster.
+- A parameter group.
+- A subnet group, based on the subnet ids provided.
+- A Security group, based on the VPC CIDR provided.
+- Two Security group rules,an egress and an ingress rule.
+
+## Inputs
+**Important**
+All inputs :heavy_check_mark: must be configured.
+Any with :x: can be ignored, but can be configurd if we want.
+
+| Name        | Description | Required | Type | Default |
+| ----------- | ----------- | -------- | ---- | ------- |
+| name | All named components will have this variable added as a suffix | :heavy_check_mark: | string | "common" |
+| environment | Environment name | :x: | string | "dev" |
+| vpc_id | The VPC id in which the cache need to be created | :heavy_check_mark: | string | |
+| allowed_cidr| The CIDR blocks allowed in Security group | :heavy_check_mark: | list(string) | |
+| subnet_ids | A list of VPC subnet IDs to create the subnet group | :heavy_check_mark: | list(string) | |
+| engine | Elastic cache engine | :heavy_check_mark: | string | "memcached" |
+| engine_version | Memcached engine version. For more info, see https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/supported-engine-versions.html" | :heavy_check_mark: | string | "1.6.17" |
+| instance_type | Elastic cache instance type | :heavy_check_mark: | string | "cache.t2.micro" |
+| cluster_size | Cluster size | :heavy_check_mark: | number | 1 |
+| parameter_group_name | Name of the existing parameter group | :heavy_check_mark: | string | |
+| parameter_group_family | Name of the cache parameter group family | :heavy_check_mark: | string | "memcached1.6" |
+| maintenance_window | Maintenance window | :x: | string | |
+| apply_immediately | Specifies whether any modifications are applied immediately, or during the next maintenance window. Default is false. | :x: | bool | false |
+| notification_topic_arn | An Amazon Resource Name (ARN) of an SNS topic to send ElastiCache notifications to. Example: arn:aws:sns:us-east-1:xxxxxxxxxx:my_sns_topic | :x: | string | |
+| port | Memcached port | :x: | number | 11211 |
+| az_mode | Enable or disable multiple AZs, eg: single-az or cross-az | :x: | string | "single-az" |
+| availability_zone | The Availability Zone of the cluster. az_mode must be set to single-az when used. | :x: | string | |
+| availability_zones | List of Availability Zones for the cluster. az_mode must be set to cross-az when used | :x: | list(string) | |
+
+## Caveats
```

Comment:
I dont understand the caveat here? I thought the Redis module is a separate module to the memcached one?

Reply:
Please also add examples for both so that we can run test plans.

Reply:
> I dont understand the caveat here? I thought the Redis module is a separate module to the memcached one?

The Redis and Memcached module is kept separate, but using the Terraform resource "aws_elasticache_cluster" used in the Memcached module, we can create both Memcached cluster and Single-node Redis instance (Cluster mode disabled, no read replica) as well. 
In the Redis module, the Terraform resource "aws_elasticache_replication_group" is used, which creates Redis cluster with read replicas (in both Cluster mode enabled and disabled)
Reference:
https://registry.terraform.io/providers/figma/aws-4-49-0/latest/docs/resources/elasticache_cluster
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_replication_group.html

Reply:
> Please also add examples for both so that we can run test plans.

Gave example as values for variable required other than the default values given for remaining variables which can be used to run this.

Reply:
> > I dont understand the caveat here? I thought the Redis module is a separate module to the memcached one?
> 
> The Redis and Memcached module is kept separate, but using the Terraform resource "aws_elasticache_cluster" used in the Memcached module, we can create both Memcached cluster and Single-node Redis instance (Cluster mode disabled, no read replica) as well. In the Redis module, the Terraform resource "aws_elasticache_replication_group" is used, which creates Redis cluster with read replicas (in both Cluster mode enabled and disabled) Reference: https://registry.terraform.io/providers/figma/aws-4-49-0/latest/docs/resources/elasticache_cluster https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticache_replication_group.html

Modifications done:
I have now separated redis and memcached modules. In redis module, we can create both replication group or single node redis cluster according to our requirement. Also, updated README file.